src/main/java/nl/wearefrank/openapifrankadapter/OpenapiFrankadapterApplication.java

```diff
@@ -56,41 +56,75 @@ public static void main(String[] args) {
     @PostMapping(value = "/receiver-file", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
     public ResponseEntity<Resource> postFileReceiver(@RequestParam("file") MultipartFile file) throws IOException, SAXException {
         // Check if it's a JSON file
-        if (!file.getContentType().equals("application/json") && !file.getContentType().equals("application/yaml")) {
+        if (!file.getContentType().matches("application/(json|yaml|x-yaml|octet-stream|text/yaml|plain)")) {
             return ResponseEntity.status(415)
                     .contentType(MediaType.APPLICATION_JSON)
                     .body(new InputStreamResource(new ByteArrayInputStream("{\"message\": \"Unsupported Media Type\"}".getBytes())));
         } else {
-            GenFiles convertedFile = new GenFiles("inputed-api.json", file.getBytes());
+            GenFiles convertedFile;
+            if (!file.getContentType().equals("application/json"))
+                convertedFile = new GenFiles("inputted-api.yaml", file.getBytes());
+            else {
+                convertedFile = new GenFiles("inputted-api.json", file.getBytes());
+            }
             return responseGenerator(convertedFile, Option.RECEIVER);
         }
     }
 
     @PostMapping(value = "/receiver-url", consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
     public ResponseEntity<Resource> postUrlReceiver(@RequestParam("url") String url) throws IOException, SAXException {
-        GenFiles convertedFile = new GenFiles("inputed-api.json", downloadFileFromUrl(url));
+        GenFiles convertedFile = new GenFiles("inputted-api.json", downloadFileFromUrl(url));
         return responseGenerator(convertedFile, Option.RECEIVER);
     }
 
     @PostMapping(value = "/sender-file", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
     public ResponseEntity<Resource> postFileSender(@RequestParam("file") MultipartFile file) throws IOException, SAXException {
         // Check if it's a JSON file
```

Comment:
Dit is weer hetzelfde stukje. Kan je de content type check in een aparte method kwijt? Dan wordt het iets als `if (isSupportedContentType(file.getContentType())` oid